### PR TITLE
feat: add config validation to resolveOptions

### PR DESCRIPTION
**Pull request by maxence2997**
## Summary
- Added `validateOptions()` with 15 individual checks matching client-kt reference
- All messages use `wspulse:` prefix for cross-SDK consistency
- Validates: maxMessageSize, writeWait, heartbeat (pingPeriod, pongWait, pingPeriod < pongWait), autoReconnect (maxRetries, baseDelay, maxDelay)
- 20 new test cases in options.test.ts

## Test plan
- [x] All 20 validation tests pass
- [x] Existing 6 options tests still pass
- [x] `make check` passes (prettier + eslint + tsc + vitest)

### CODE REVIEW

File: src/options.ts [hunk 1]
```diff
@@ -131,6 +140,74 @@ export interface ResolvedOptions {
...
+  }
+
+  if (opts.writeWait !== undefined) {
+    if (opts.writeWait <= 0) {
+      throw new Error("wspulse: writeWait must be positive");
+    }
+    if (opts.writeWait > MAX_WRITE_WAIT) {
+      throw new Error("wspulse: writeWait exceeds maximum (30s)");
+    }
+  }
```
[the remark this answers is not in the document]
Comment (maxence2997) [reply src/options.ts]: **Not adjusted.** TypeScript callers get compile-time type safety — `NaN` would require deliberately bypassing the type system. Adding `Number.isFinite` guards for JS runtime safety is a valid concern but adds complexity that does not benefit the primary TS audience. May revisit if JS usage grows.

File: src/options.ts [hunk 2]
```diff
@@ -131,6 +140,74 @@ export interface ResolvedOptions {
...
+      throw new Error("wspulse: autoReconnect.baseDelay exceeds maximum (1m)");
+    }
+    if (rc.maxDelay < rc.baseDelay) {
+      throw new Error("wspulse: autoReconnect.maxDelay must be >= baseDelay");
+    }
+    if (rc.maxDelay > MAX_MAX_DELAY) {
+      throw new Error("wspulse: autoReconnect.maxDelay exceeds maximum (5m)");
+    }
+    if (rc.maxRetries > 0 && rc.maxRetries > MAX_MAX_RETRIES) {
+      throw new Error("wspulse: autoReconnect.maxRetries exceeds maximum (32)");
```
[the remark this answers is not in the document]
Comment (maxence2997) [reply src/options.ts]: **Not adjusted.** Same as above — TypeScript type system already prevents this.

File: test/options.test.ts [hunk 3]
```diff
@@ -83,3 +83,159 @@ describe("resolveOptions", () => {
...
+  it("accepts max boundary values", () => {
+    expect(() =>
+      resolveOptions({
+        maxMessageSize: 64 << 20,
+        writeWait: 30_000,
+        heartbeat: { pingPeriod: 59_000, pongWait: 120_000 },
+        autoReconnect: {
+          maxRetries: 32,
+          baseDelay: 60_000,
+          maxDelay: 300_000,
```
[the remark this answers is not in the document]
Comment (maxence2997) [reply test/options.test.ts]: **Fixed in 95164c6.** Updated `pingPeriod` from 59,000 to 60,000 to test the inclusive maximum boundary.

File: src/options.ts [hunk 4]
```diff
@@ -131,6 +140,74 @@ export interface ResolvedOptions {
...
+    if (rc.maxDelay < rc.baseDelay) {
+      throw new Error("wspulse: autoReconnect.maxDelay must be >= autoReconnect.baseDelay");
+    }
+    if (rc.maxDelay > MAX_MAX_DELAY) {
+      throw new Error("wspulse: autoReconnect.maxDelay exceeds maximum (5m)");
+    }
+    if (rc.maxRetries > 0 && rc.maxRetries > MAX_MAX_RETRIES) {
+      throw new Error("wspulse: autoReconnect.maxRetries exceeds maximum (32)");
+    }
+  }
```
[the remark this answers is not in the document]
Comment (maxence2997) [reply src/options.ts]: **Not adjusted.** TypeScript requires all fields in `HeartbeatOptions` and `AutoReconnectOptions` — omitting `pingPeriod` or `pongWait` is a compile error. This is a JS-only concern that adds complexity for the non-target audience.

File: src/options.ts [hunk 5]
```diff
@@ -10,7 +10,7 @@ import { JSONCodec } from "./codec.js";
  * transport drops, using exponential backoff with equal jitter.
  */
 export interface AutoReconnectOptions {
-  /** Max retry attempts. 0 or negative = unlimited. */
+  /** Max retry attempts. 0 = unlimited. Must be non-negative. */
```
[the remark this answers is not in the document]
Comment (maxence2997) [reply src/options.ts]: **Already addressed.** The doc comment was updated in the validation commit from `0 or negative = unlimited` to `0 = unlimited. Must be non-negative.`
